### PR TITLE
Feature: Payment

### PR DESCRIPTION
**Pull request by aniansson**
## Pivotal: 
https://www.pivotaltracker.com/story/show/137258649

## Changes proposed in this pull request: 
Add a payment feature with help of stripe and shopping cart. Add API key and gems.

## What I have learned working on this feature: 
How to implement stripe, shopping carts. As well as repeating the knowledge of routes, models, migrating and databases, as well as gems, capybara and poltergeist. Setting up API keys and how to hide them fro posting on github. 

### Checkout screen: 
screen shot 2017-01-12 at 16 08 11

### After clicking "Pay with Card" button
screen shot 2017-01-12 at 16 08 46

### After you click the "Pay" after the form and the payment goes through.
screen shot 2017-01-12 at 16 08 58

Ready for review!

### CODE REVIEW

File: app/views/layouts/charges.html.erb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<!DOCTYPE html>
+<html>
+  <head></head>
+  <body>
+    <%= yield %>
+  </body>
+</html>
```
Comment (tochman) [inline app/views/layouts/charges.html.erb:7]: Why do you create a separate layout file? Please explain.

File: config/routes.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
   root controller: :landing, action: :index
   resources :dishes, only: [:index, :show]
+  resources :charges
```
Comment (tochman) [inline config/routes.rb]: This creates a lot of routes, do you really need them?

**Comment by Philippoes** [comment]
# Screenshots: 
### Checkout screen: 
screen shot 2017-01-12 at 16 08 11

### After clicking "Pay with Card" button
screen shot 2017-01-12 at 16 08 46

### After you click the "Pay" after the form and the payment goes through.
screen shot 2017-01-12 at 16 08 58

File: app/controllers/landing_controller.rb [hunk 3]
```diff
@@ -2,4 +2,12 @@ class LandingController < ApplicationController
   def index
     @dishes = Dish.all
   end
+
+  def checkout
+    @total_amount = 50
+    # ShoppingCart.find_by(id: 1).shopping_cart_items.each do |shoppingcartitem|
+    #   @totalamount += shoppingcartitem.item.price
+    # end
```
Comment (AmberWilkie) [inline app/controllers/landing_controller.rb]: Delete comments, please.

**Review by AmberWilkie — APPROVED** [review]

**Comment by AmberWilkie** [comment]
@aniansson I don't know why Travis isn't kicking in. Delete those comments and push this again to see if that makes it start. Sometimes these external services can be a little buggy.

Make sure you pull from origin first.

**Pull request merged**